index.d.ts

```diff
@@ -1,38 +1,38 @@
-import { Middleware, MiddlewareAPI } from 'redux';
+import { Middleware, MiddlewareAPI, Action } from 'redux';
 import { Observable, ObservableInput } from 'rxjs/Observable';
 import { Scheduler } from 'rxjs/Scheduler';
 import { Operator } from 'rxjs/Operator';
 
-export declare class ActionsObservable<T> extends Observable<T> {
+export declare class ActionsObservable<T extends Action> extends Observable<T> {
   /**
    * Just like RxJS itself, we can't actually make this method always type-safe
    * because we would need non-final position spread params e.g.
    *   `static of<T>(...items: T, scheduler?: Scheduler): ActionsObservable<T>`
    * which isn't possible in either JavaScript or TypeScript. So instead, we
    * provide safe typing for up to 6 items, following by a scheduler.
    */
-  static of<T>(item1: T, scheduler?: Scheduler): ActionsObservable<T>;
-  static of<T>(item1: T, item2: T, scheduler?: Scheduler): ActionsObservable<T>;
-  static of<T>(item1: T, item2: T, item3: T, scheduler?: Scheduler): ActionsObservable<T>;
-  static of<T>(item1: T, item2: T, item3: T, item4: T, scheduler?: Scheduler): ActionsObservable<T>;
-  static of<T>(item1: T, item2: T, item3: T, item4: T, item5: T, scheduler?: Scheduler): ActionsObservable<T>;
-  static of<T>(item1: T, item2: T, item3: T, item4: T, item5: T, item6: T, scheduler?: Scheduler): ActionsObservable<T>;
-  static of<T>(...array: Array<T | Scheduler>): ActionsObservable<T>;
+  static of<T extends Action>(item1: T, scheduler?: Scheduler): ActionsObservable<T>;
+  static of<T extends Action>(item1: T, item2: T, scheduler?: Scheduler): ActionsObservable<T>;
+  static of<T extends Action>(item1: T, item2: T, item3: T, scheduler?: Scheduler): ActionsObservable<T>;
+  static of<T extends Action>(item1: T, item2: T, item3: T, item4: T, scheduler?: Scheduler): ActionsObservable<T>;
+  static of<T extends Action>(item1: T, item2: T, item3: T, item4: T, item5: T, scheduler?: Scheduler): ActionsObservable<T>;
+  static of<T extends Action>(item1: T, item2: T, item3: T, item4: T, item5: T, item6: T, scheduler?: Scheduler): ActionsObservable<T>;
+  static of<T extends Action>(...array: Array<T | Scheduler>): ActionsObservable<T>;
 
-  static from<T>(ish: ObservableInput<T>, scheduler?: Scheduler): ActionsObservable<T>;
-  static from<T, R>(ish: ArrayLike<T>, scheduler?: Scheduler): ActionsObservable<R>;
+  static from<T extends Action>(ish: ObservableInput<T>, scheduler?: Scheduler): ActionsObservable<T>;
+  static from<T extends Action, R extends Action>(ish: ArrayLike<T>, scheduler?: Scheduler): ActionsObservable<R>;
 
   constructor(input$: Observable<T>);
-  lift<R>(operator: Operator<T, R>): ActionsObservable<R>;
-  ofType(...key: string[]): ActionsObservable<T>;
-  ofType(...key: any[]): ActionsObservable<T>;
+  lift<R extends Action>(operator: Operator<T, R>): ActionsObservable<R>;
+  lift<R>(operator: Operator<T, R>): Observable<R>;
```

Comment:
This should still return and `ActionsObservable` (the same signature as it was before) or am I confused?

```ts
lift<R>(operator: Operator<T, R>): ActionsObservable<R>;
```

It's not super useful right now since it would only give you chaining `ofTypes()`, but later we may add additional operators besides `ofType`. e.g. #281

```ts
action$
  .ofType(SOMETHING)
  .debounceTime(200)
  .anotherCustomOperatorWeProvide()
```